app/Kwality.UVault.QA/M2M/Application.Token.Management.Auth0.Tests.cs

```diff
@@ -0,0 +1,127 @@
+// =====================================================================================================================
+// = LICENSE:       Copyright (c) 2023 Kevin De Coninck
+// =
+// =                Permission is hereby granted, free of charge, to any person
+// =                obtaining a copy of this software and associated documentation
+// =                files (the "Software"), to deal in the Software without
+// =                restriction, including without limitation the rights to use,
+// =                copy, modify, merge, publish, distribute, sublicense, and/or sell
+// =                copies of the Software, and to permit persons to whom the
+// =                Software is furnished to do so, subject to the following
+// =                conditions:
+// =
+// =                The above copyright notice and this permission notice shall be
+// =                included in all copies or substantial portions of the Software.
+// =
+// =                THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// =                EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// =                OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// =                NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// =                HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// =                WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// =                FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// =                OTHER DEALINGS IN THE SOFTWARE.
+// =====================================================================================================================
+namespace Kwality.UVault.QA.M2M;
+
+using FluentAssertions;
+
+using global::System.Diagnostics.CodeAnalysis;
+
+using JetBrains.Annotations;
+
+using Kwality.UVault.Auth0.Keys;
+using Kwality.UVault.Auth0.M2M.Configuration;
+using Kwality.UVault.Auth0.M2M.Extensions;
+using Kwality.UVault.Auth0.M2M.Mapping.Abstractions;
+using Kwality.UVault.Auth0.M2M.Models;
+using Kwality.UVault.Auth0.Models;
+using Kwality.UVault.M2M.Managers;
+using Kwality.UVault.M2M.Models;
+using Kwality.UVault.QA.Internal.Factories;
+using Kwality.UVault.QA.Internal.System;
+using Kwality.UVault.QA.Internal.Xunit.Traits;
+
+using Xunit;
+
+public sealed class ApplicationTokenManagementAuth0Tests
```

Comment:
The tests here aren't adequate enough.
Basically, what you're testing is if you can retrieve an access management token, but not a token for a M2M application.

I guess the test should rather be:

- Create a new application (using already existing functionality).
- Verify if you can request a token for that application.

This is what you need to do for all the tests (Auth0, Custom and Static).